nucleus/dataset.py

```diff
@@ -227,17 +236,29 @@ def list_autotags(self):
 
     def _format_dataset_item_response(self, response: dict) -> dict:
         item = response.get(ITEM_KEY, None)
-        annotation_payload = response.get(ANNOTATIONS_KEY, [])
+        annotation_payload = response.get(ANNOTATIONS_KEY, {})
         if not item or not annotation_payload:
             # An error occured
             return response
-        annotations = [
-            BoxAnnotation.from_json(ann)
-            if ann["type"] == "box"
-            else PolygonAnnotation.from_json(ann)
-            for ann in annotation_payload
-        ]
+
+        annotation_response = {}
+        if BOX_TYPE in annotation_payload:
```

Comment:
Could consider an outer for loop like

```python
for (type_key, type_cls) in zip([BOX_TYPE, POLYGON_TYPE, SEGMENTATION_TYPE],
        [BoxAnnotation, PolygonAnnotation, SegmentationAnnotation]):
```

to avoid the copypaste if desired, though I'm not sure that's much better.

Best would be if we added a base `Annotation` parent class in annotation.py which implemented the `.from_json` class method itself (which routed to the appropriate subclass's `.from_json` depending on the value of the type key in the json) to return the correct subclass instance. And added a `ANNOTATION_TYPES=(BOX_TYPE, POLYGON_TYPE, SEGMENTATION_TYPE)` tuple to constants.py. Then this implementation could look like just:

```python
for type in ANNOTATION_TYPES:
    if type in annotation_payload:
        annotation_response[type] = [Annotation.from_json(ann) for ann in annotation_payload[type]]
```